File: plugin/src/main/java/com/stylegenerator/plugin/util/ResourceReader.java
package com.stylegenerator.plugin.util;

import java.io.File;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public final class ResourceReader {

    private static final String FONTS_FOLDER = "src/main/res/font";
    private static final String COLORS_FILE = "src/main/res/values/colors.xml";

    private static final String EXTENSION_SEPARATOR = ".";

    private ResourceReader() {
    }

    public static List<String> getFontNames() {
        List<String> fontsWithoutExtension = Collections.emptyList();

        String[] fonts = new File(FONTS_FOLDER).list();
        if (fonts != null && fonts.length > 0) {
            fontsWithoutExtension = Arrays.stream(fonts)
                    .map(filename -> filename.substring(0, filename.lastIndexOf(EXTENSION_SEPARATOR)))
                    .collect(Collectors.toList());
        }

        return fontsWithoutExtension;
    }

    public static List<String> getColorNames() {
        Map<String, String> colors = XMLHelper.readFromXML(new File(COLORS_FILE), "color");
        return new ArrayList<>(colors.keySet());
    }
}

File: plugin/src/main/java/com/stylegenerator/plugin/util/ResourceReader.kt
package com.stylegenerator.plugin.util

import java.util.ArrayList

object ResourceReader {

    private const val FONTS_FOLDER = "src/main/res/font"
    private const val COLORS_FILE = "src/main/res/values/colors.xml"

    private const val EXTENSION_SEPARATOR = "."

    fun getFontNames(path: String): List<String> {
        val fonts = FileUtil.getFile(path, FONTS_FOLDER).list()
        return when {
            fonts.isNullOrEmpty() -> emptyList()
            else -> fonts.map { filename ->
                filename.substring(0, filename.lastIndexOf(EXTENSION_SEPARATOR))
            }
        }
    }

    fun getColorNames(path: String): List<String> {
        val colors = XMLUtil.readFromXML(FileUtil.getFile(path, COLORS_FILE), "color")
        return ArrayList(colors.keys)
    }
}
